src/main/java/org/opentripplanner/model/plan/legreference/LegReferenceType.java

```diff
@@ -0,0 +1,59 @@
+package org.opentripplanner.model.plan.legreference;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.text.ParseException;
+
+/**
+ * Enum for different types of LegReferences
+ */
+enum LegReferenceType {
+  SCHEDULED_TRANSIT_LEG_V1(
```

Comment:
V1 = version 1? How will the versioning work here?

Reply:
We can have different types of references, and they can have different versions. Eg. we could add `FREQUENCY_TRANSIT_LEG_V1 ` and `FLEX_TRANSIT_LEG_V1` now, and if we at a later stage find out we need to adjust the serialization of the flex leg, we can add a `FLEX_TRANSIT_LEG_V2`, with updated serialization with the missing information.